src/schema/mbs_types.ts

```diff
@@ -0,0 +1,267 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import _ from 'lodash';
+import debug from 'debug';
+import * as graphql from 'graphql';
+import {
+  GraphQLBoolean,
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from 'graphql';
+import { delegateToSchema } from '@graphql-tools/delegate';
+
+import schema from './schema';
+import { CommitObject, DistGitInstanceInputType } from './distgit_types';
+import {
+  KojiBuildTagsType,
+  KojiHistoryType,
+  KojiInstanceInputType,
+} from './koji_types';
+
+const log = debug('osci:mbs_types');
+
+export interface MbsTaskFields {
+  component: string;
+  id?: number;
+  nvr: string;
+  state: number;
+}
+
+export const MbsTaskType = new GraphQLObjectType<MbsTaskFields, {}>({
+  name: 'MbsTaskType',
+  fields: {
+    component: {
+      description: 'Name of the component comprising the task',
+      type: GraphQLNonNull(GraphQLString),
+    },
+    id: { description: 'ID of the Koji task', type: GraphQLInt },
+    nvr: {
+      description: 'Full NVR of the comprising package',
+      type: GraphQLNonNull(GraphQLString),
+    },
+    state: {
+      description: 'Number of the state the task is currently in',
+      type: GraphQLNonNull(GraphQLInt),
+    },
+  },
+});
+
+export interface MbsBuildFields {
+  commit: any;
+  context: string;
+  id: number;
+  koji_tag: string;
+  name: string;
+  owner: string;
+  scmurl: string;
+  scratch: boolean;
+  stream: string;
+  tasks: MbsTaskFields[];
+  time_completed?: string;
+  version: string;
+}
+
+interface MbsBuildCommitArgs {
+  instance: string;
+}
+
+/*
+ * TOOD: The third type argument should really be `MbsBuildCommitArgs`, but using
+ * that causes a type error further down below in when passing the resolver to the
+ * `resolve` field of `commit` within `MbsBuildType`.
+ */
+const commitResolver: graphql.GraphQLFieldResolver<any, {}, any> = async (
+  parentValue,
+  args,
+  context,
+  info,
+) => {
+  const { scmurl } = parentValue;
+  const { instance } = args;
+  const nameWithCommit = _.last(_.split(scmurl, 'modules/'));
+  const [repoName, sha1] = _.split(nameWithCommit, '?#');
+  log('Getting commit object for %s:%s', repoName, sha1);
+  if (!repoName || !sha1) {
+    return {};
+  }
+  return await delegateToSchema({
+    schema: schema,
+    operation: 'query',
+    fieldName: 'distgit_commit',
+    args: {
+      commit_sha1: sha1,
+      instance,
+      namespace: 'modules',
+      repo_name: repoName,
+    },
+    context,
+    info,
+  });
+};
+
+const tagHistoryResolver: graphql.GraphQLFieldResolver<any, {}, any> = async (
+  parentValue,
+  args,
+  schemaContext,
+  info,
+) => {
+  const { context, name, stream, version } = parentValue;
+  const { instance } = args;
+  const nsvc = `${name}-${stream}-${version}.${context}`;
```

Comment:
You construct here NVR based on NSVC values.
Var name should be `nvr`.
https://docs.fedoraproject.org/en-US/modularity/core-concepts/nsvca/#_binary_level_id_nsvca
`Yum` command taskes NSVC in format: N:S:V:C:A
another place for reference : https://pagure.io/fedora-ci/messages/blob/master/f/schemas/module-build.yaml#_50

Reply:
Didn't realise that, thanks for the references.